src/main/java/com/minecolonies/coremod/util/BarbarianUtils.java

```diff
@@ -0,0 +1,195 @@
+import com.minecolonies.coremod.entity.ai.mobs.EntityBarbarian;
+import com.minecolonies.coremod.entity.ai.mobs.EntityChiefBarbarian;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityList;
+import net.minecraft.init.Items;
+import net.minecraft.inventory.EntityEquipmentSlot;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.MathHelper;
+import net.minecraft.world.World;
+
+import java.util.stream.IntStream;
+
+/**
+ * Utils for The Barbarian Raid Event
+ */
+public final class BarbarianUtils
+{
+
+    /**
+     * ResourceLocations for barbarians
+     */
+    private static final ResourceLocation barbarian = EntityList.getKey(EntityBarbarian.class);
+    private static final ResourceLocation archer    = EntityList.getKey(EntityArcherBarbarian.class);
+    private static final ResourceLocation chief     = EntityList.getKey(EntityChiefBarbarian.class);
+
+    private static final int    LEVEL_AT_WHICH_TO_NOT_TRIGGER_RAID = 1;
+    private static final int    MAX_SIZE                           = Configurations.maxBarbarianHordeSize;
+    private static final double BARBARIANS_MULTIPLIER = 0.5;
+    private static final double ARCHER_BARBARIANS_MULTIPLIER       = 0.25;
+    private static final double CHIEF_BARBARIANS_MULTIPLIER        = 0.1;
+    private static final int    NUMBER_OF_POSSIBLE_CASES           = 360;
+    private static final int    PREFERRED_MAX_HORDE_SIZE            = 40;
+    private static final int    PREFERRED_MAX_BARBARIANS            = 22;
+    private static final int    PREFERRED_MAX_ARCHERS               = 16;
+    private static final int    PREFERRED_MAX_CHIEFS                = 2;
+    private static final float  WHOLE_CIRCLE                       = 360.0F;
+    private static final float  HALF_A_CIRCLE                      = 180F;
+
+    /**
+     * Private constructor to hide the implicit public one.
+     */
+    private BarbarianUtils()
+    {
+    }
+
+    public static void doRaid(final World world, final int level, final Colony colony)
+    {
+        eventRaid(world, level, colony);
+    }
+
+    /**
+     * Commences a Raid event in a specific colony, with a specific RaidLevel
+     */
+
+    private static void eventRaid(final World raidingWorld, final int level, final Colony colony)
+    {
+        if (level == LEVEL_AT_WHICH_TO_NOT_TRIGGER_RAID)
+        {
+            return;
+        }
+
+        int numberOfBarbarians = (int) (BARBARIANS_MULTIPLIER * level) ;
+        int numberOfArcherBarbarians = (int) (ARCHER_BARBARIANS_MULTIPLIER * level);
+        int numberOfChiefBarbarians = (int) (CHIEF_BARBARIANS_MULTIPLIER * level);
+
+        int hordeTotal = numberOfArcherBarbarians + numberOfBarbarians + numberOfChiefBarbarians;
+
+        if (hordeTotal > PREFERRED_MAX_HORDE_SIZE && MAX_SIZE == PREFERRED_MAX_HORDE_SIZE && hordeTotal > MAX_SIZE)
```

Comment:
you don't need to do this test **hordeTotal > MAX_SIZE** this is te consequence of you 2 previous conditions